### PR TITLE
CSP-1722: Upgrade SFA.DAS.Fat.DataLoader to .Net 8.0

### PR DESCRIPTION
**Pull request by shaun-downey-education**
- Upgrade azure function project in **SFA.DAS.DataLoader** to .Net 8.0.
- Updated README file incorporate correct **local.settings.json**.
- Updated tag reference to newest version.

### CODE REVIEW

**Review by debrr07 — APPROVED** [review]

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.Functions.Importer/host.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    "version": "2.0",
-    "logging": {
-        "applicationInsights": {
-            "samplingExcludedTypes": "Request",
-            "samplingSettings": {
-                "isEnabled": true
-            }
-        }
-    },
-    "functionTimeout": "00:35:00"
```
Comment (DevenKShah) [inline src/SFA.DAS.Functions.Importer/host.json:11]: I think we need to keep function time out

File: src/SFA.DAS.Functions.Importer/Extensions/AddConfigurationExtension.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        builder
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("local.settings.json", optional: true);
+
+        var config = builder.Build();
+
+        builder.AddAzureTableStorage(options =>
+        {
+            options.ConfigurationKeys = config["ConfigNames"].Split(",");
```
Comment (DevenKShah) [inline src/SFA.DAS.Functions.Importer/Extensions/AddConfigurationExtension.cs:19]: There is a compile time warning here. Please make sure there are no compile time warnings.

File: src/SFA.DAS.Functions.Importer/Extensions/AddConfigurationExtension.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.Extensions.Configuration;
+using SFA.DAS.Configuration.AzureTableStorage;
+using System.IO;
```
Comment (DevenKShah) [inline src/SFA.DAS.Functions.Importer/Extensions/AddConfigurationExtension.cs:3]: unused using

File: src/SFA.DAS.Functions.Importer/Extensions/AddServiceRegistrationsExtension.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace SFA.DAS.Functions.Importer.Extensions;
+
+public static class AddServiceRegistrationsExtension
+{
+    public static IServiceCollection AddServiceRegistrations(this IServiceCollection services, IConfiguration configuration)
+    {
+        services
+            .RegisterServices()
+            .AddHttpClient()
```
Comment (DevenKShah) [inline src/SFA.DAS.Functions.Importer/Extensions/AddServiceRegistrationsExtension.cs:18]: This is redundant as it is invoked in the method as well

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**